File: problems/2103-rings-and-rods/solution.java
class Solution {
    public int countPoints(String rings) {
        int[] rods = new int[10];
        for (int i = 0; i < rings.length(); i += 2) {
            char color = rings.charAt(i);
            int rod = rings.charAt(i + 1) - '0';
            if (color == 'R') {
                rods[rod] |= 1;
            } else if (color == 'G') {
                rods[rod] |= 2;
            } else {
                rods[rod] |= 4;
            }
        }
        int count = 0;
        for (int i = 0; i < 10; i++) {
            if (rods[i] == 7) {
                count++;
            }
        }
        return count;
    }
}

File: problems/2103-rings-and-rods/solution.py
class Solution:
    def countPoints(self, rings: str) -> int:
        rods = [set() for _ in range(10)]
        for i in range(0, len(rings), 2):
            color = rings[i]
            rod_num = int(rings[i+1])
            rods[rod_num].add(color)
        count = 0
        for rod in rods:
            if len(rod) == 3:
                count += 1
        return count